### PR TITLE
Resize SVG element for donut chart model using JS and CSS

### PR DESCRIPTION
**Pull request by gabycampagna**
screen-shot-2017-06-05-at-7 59 43-am
screen-shot-2017-06-05-at-8 02 11-am

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit 3389e02c6438a7ee188500994d112a70325b3cdc

https://deploy-preview-3015--ilios-frontend.netlify.com

File: app/components/chart-donut.js [hunk 1]
```diff
@@ -43,6 +43,14 @@ export default Component.extend({
       return;
     }
 
+    if (!isIcon) {
+      svg.classed("svg-container", true)
```
Comment (jrjohnson) [inline app/components/chart-donut.js]: As I understand this code the class `.svg-container` gets added here, is that right?

File: app/components/chart-donut.js [hunk 2]
```diff
@@ -43,6 +43,14 @@ export default Component.extend({
...
+      .attr("preserveAspectRatio", "xMinYMin meet")
+      .attr("viewBox","0 0 " + width + " " + height)
+      //class to make it responsive
+      .classed("svg-content-responsive", true);
```
Comment (jrjohnson) [inline app/components/chart-donut.js]: Do we need both of these classes? Shouldn't we just make all SVGs responsive? Or put another way - is there an `.svg-container` that isn't also going to be responsive?

File: app/styles/newcomponents/ilios-chart.scss [hunk 3]
```diff
@@ -3,5 +3,37 @@
...
+    }
+  }
+
+  .svg-container {
+    overflow: visible;
+    padding-bottom: 100%; // aspect ratio
+    vertical-align: top;
+  }
+
+  @include media($small-screen) {
```
Comment (jrjohnson) [inline app/styles/newcomponents/ilios-chart.scss]: These media queries feel a bit inverted to me can this be written as 
```
.svg-container {
  overflow: visible;
  padding-bottom: 100%; // aspect ratio
  vertical-align: top;

  @include media($small-screen) {
    padding-bottom: 26%;
    width: 50%;
  }
}
```

File: app/components/chart-donut.js (hunk 1 above)
Comment (gabycampagna) [inline app/components/chart-donut.js]: Yes, but.. It should actually be added to a parent div or other container.  Wasn't sure if that is the route we want to go, so I added it to the svg element.  Please let me know you thoughts about that.

File: app/components/chart-donut.js (hunk 2 above)
Comment (gabycampagna) [inline app/components/chart-donut.js]: We could or "should" make all SVG responsive, with the exception of the Icons of course.  We could add a parent container to the SVG as I mentioned in the comment above this one.

File: app/components/chart-donut.js (hunk 1 above)
Comment (jrjohnson) [inline app/components/chart-donut.js]: Ah ok. Yes. Maybe we can apply all of this to the ilios-chart component. I'm thinking:
1) Get rid of width / height properties on all chart components
2) Handle size through CSS completely
3) Keep isIcon and just apply that as a class on the ilios-chart component.

Will that work with the D3 svg? I'm basically just wishing and making stuff up so let me know if that wont work at all.

Comment (gabycampagna) [inline app/components/chart-donut.js]: I will give it a try and let you know!  But I think the currently solution we have is the simplest one. I will think this through and get back to you.

**Comment by jrjohnson** [comment]
@dartajax please review animations throughout the application to ensure they aren't wonky. Then this can be merged.

**Pull request merged**